### ISSUE TITLE
Add a rule for no-trailing-blank-lines for docstrings

### ISSUE THREAD

**Issue by chazmo03** [astral-sh/ruff#19564]
### Summary

Trailing blank lines _within_ multi-line docstrings should be flagged and removed. This already happens for single-line docstrings (thanks to `unnecessary-multiline-docstring`), so it seems sensible to do the same for multi-line docstrings as well.

See example:

```py
def function() -> None:
    """This single-line docstring would be fixed to remove the trailing blank lines.


    """

def function() -> None:
    """This multi-line docstring would NOT be fixed to remove the trailing blank lines.

    So long as there was a second non-blank line.


    """

```

**Comment by ntBre** [comment]
That makes sense to me. Interestingly, it looks like the formatter will also perform the first fix: https://play.ruff.rs/1e04904d-d7e3-4c3d-a71d-1965834e77c0?secondary=Format

but not the second.

**Comment by MichaReiser** [comment]
We might also want to consider adding this behavior to an existing rule, e.g. https://docs.astral.sh/ruff/rules/surrounding-whitespace/

**Comment by chazmo03** [comment]
@MichaReiser combining it into `surrounding-whitespace (D210)` makes sense to me.
